client/src/utils/__tests__/jsonUtils.test.ts

```diff
@@ -317,3 +317,112 @@ describe("getValueAtPath", () => {
     expect(getValueAtPath(obj, ["users", "1", "name"])).toBe("Jane");
   });
 });
+
+describe("JsonSchemaType elicitation field support", () => {
+  const sampleSchema: JsonSchemaType = {
+    type: "object",
+    title: "User Info",
+    description: "User information form",
+    properties: {
+      name: {
+        type: "string",
+        title: "Full Name",
+        description: "Your full name",
+        minLength: 2,
+        maxLength: 50,
+        pattern: "^[A-Za-z\\s]+$",
+      },
+      email: {
+        type: "string",
+        format: "email",
+        title: "Email Address",
+      },
+      age: {
+        type: "integer",
+        minimum: 18,
+        maximum: 120,
+        default: 25,
+      },
+      role: {
+        type: "string",
+        enum: ["admin", "user", "guest"],
+        enumNames: ["Administrator", "User", "Guest"],
+      },
+    },
+    required: ["name", "email"],
+  };
+
+  test("should parse JsonSchemaType with elicitation fields", () => {
+    const schemaString = JSON.stringify(sampleSchema);
+    const result = tryParseJson(schemaString);
+
+    expect(result.success).toBe(true);
+    expect(result.data).toEqual(sampleSchema);
+  });
+
+  test("should update schema properties with new validation fields", () => {
+    const updated = updateValueAtPath(
+      sampleSchema,
+      ["properties", "name", "minLength"],
+      5,
+    );
+
+    expect(getValueAtPath(updated, ["properties", "name", "minLength"])).toBe(
+      5,
+    );
+  });
+
+  test("should handle enum and enumNames fields", () => {
+    const schema = {
+      type: "string" as const,
+      enum: ["option1", "option2"],
+      enumNames: ["Option 1", "Option 2"],
```

Comment:
`enumNames` [is not part of the JSONSchema spec](https://github.com/rjsf-team/react-jsonschema-form/issues/532). The [recommended way to describe](https://github.com/json-schema-org/json-schema-spec/issues/57) enum labels is with `oneOf` or `anyOf`.